applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -231,220 +206,406 @@ fn stringify_request(buffer: &[u8]) -> String {
+            );
+        },
+    }
+    data.to_vec()
+}
+
+fn handle_post(
+    method: &str,
+    json: &[u8],
+    data: Vec<u8>,
+    transient: &mut TransientData,
+    consensus: ConsensusManager,
+    rt: &mut Runtime,
+    config: &GlobalConfig,
+) -> Vec<u8>
+{
+    let grpcclientresult = rt.block_on(grpc::base_node_client::BaseNodeClient::connect(format!(
+        "{}{}",
+        "http://",
+        config.grpc_address.to_string()
+    )));
+    let mut current_data = data.clone();
+    match method {
+        "submitblock" => match grpcclientresult {
+            Ok(mut grpcclient) => {
+                let response = json::parse(&String::from_utf8_lossy(&current_data).to_string());
+                match response {
+                    Ok(response) => {
                         if response["result"]["status"] == "OK" {
                             if transient.tari_block.is_some() && transient.monero_seed.is_some() {
-                                let blockresult: grpc::GetNewBlockResult = transient
-                                    .clone()
-                                    .tari_block
-                                    .ok_or_else(|| MmProxyError::MissingDataError("New Block result".to_string()))?;
-                                let mut block: grpc::Block = blockresult
-                                    .block
-                                    .ok_or_else(|| MmProxyError::MissingDataError("Block".to_string()))?;
+                                let blockresult: grpc::GetNewBlockResult = transient.clone().tari_block.unwrap();
+                                let mut block: grpc::Block = blockresult.block.unwrap();
                                 match get_monero_data(&json, transient.clone().monero_seed.unwrap()) {
-                                    Ok(pow_data) => {
-                                        let mut tariheader: grpc::BlockHeader = block
-                                            .header
-                                            .ok_or_else(|| MmProxyError::MissingDataError("Header".to_string()))?;
-                                        let mut powdata = tariheader
-                                            .pow
-                                            .clone()
-                                            .ok_or_else(|| MmProxyError::MissingDataError("Pow data".to_string()))?;
-                                        powdata.pow_data = bincode::serialize(&pow_data).map_err(|_| {
-                                            MmProxyError::ParseError("Failure to serialize block".to_string())
-                                        })?;
+                                    Some(pow_data) => {
+                                        let mut tariheader: grpc::BlockHeader = block.header.unwrap();
+                                        let mut powdata = tariheader.pow.clone().unwrap();
+                                        powdata.pow_data = bincode::serialize(&pow_data).unwrap();
```

Comment:
We need to replace these unwraps. 
If there is a problem with malformed data, the thread will panic here

Reply:
Removed unwraps.